controllers/clickhouseuser_controller.go

```diff
@@ -4,67 +4,151 @@ package controllers
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"strconv"
 
 	avngen "github.com/aiven/go-client-codegen"
 	"github.com/aiven/go-client-codegen/handler/clickhouse"
 	"github.com/aiven/go-client-codegen/handler/service"
+	"github.com/go-logr/logr"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	ctrl "sigs.k8s.io/controller-runtime"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	"github.com/aiven/aiven-operator/api/v1alpha1"
 )
 
-// ClickhouseUserReconciler reconciles a ClickhouseUser object
-type ClickhouseUserReconciler struct {
-	Controller
+//+kubebuilder:rbac:groups=aiven.io,resources=clickhouseusers,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=aiven.io,resources=clickhouseusers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=aiven.io,resources=clickhouseusers/finalizers,verbs=get;create;update
+
+// ClickhouseUserAivenClient reconciles a ClickhouseUser object
+type ClickhouseUserAivenClient struct {
+	client.Client
+
+	Scheme         *runtime.Scheme
+	Recorder       record.EventRecorder
```

Comment:
`ClickhouseUserAivenClient` still depends on `Recorder`, but ideally event emission should live in the generic `Reconciler[T]` instead of the resource-specific Aiven client.